### PR TITLE
ComponentGenerator to add ComponentSupplier only when needed

### PR DESCRIPTION
**Pull request by asashour**
Otherwise we will have something like:

```
public class GeneratedVaadinRadioButton<R extends GeneratedVaadinRadioButton<R>>
        extends Component
        implements ComponentSupplier<R>, HasStyle, Focusable<R>, HasComponents {
```

And `ComponentSupplier` is already inherited by `Focusable`

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/flow/3018)

### CODE REVIEW

**Comment by denis-anisimov** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/flow/3018#-KzZLJEW9kAUfWOUtG-2:-KzZLJEW9kAUfWOUtG-3:bjftehg) (no related file):*
Didn't go into details in the PR but I'm against of using `AtomicBoolean` here.
The latter class is intended to be used in multithreaded code.
We use it also in unit tests but that's OK since it's just tests.

Using `AtomicBoolean` in the production code which is not intended to work in multiple threads is a bad idea.
I mostly against of the semantic. 
Though there is also performance issues since `AtomicBoolean` uses `volatile` internally. But this issue of course minor in this specific case.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3018)*

**Comment by asashour** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/flow/3018#-KzZLJEW9kAUfWOUtG-2:-KzZZPehVcElcq5J4olw:b1w4eup) (no related file):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

Didn't go into details in the PR but I'm against of using `AtomicBoolean` here.
The latter class is intended to be used in multithreaded code.
We use it also in unit tests but that's OK since it's just tests.

Using `AtomicBoolean` in the production code which is not intended to work in multiple threads is a bad idea.
I mostly against of the semantic. 
Though there is also performance issues since `AtomicBoolean` uses `volatile` internally. But this issue of course minor in this specific case.
</blockquote></details>

It was changed, so the iteration is done using `.stream()`.

However,  the second iteration would not have better performance.

Also, please note this is in Generator class, which I guess not a production environment, but is rather used to generate components java code.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3018)*

**Comment by gilberto-torrezan** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/flow/3018#-KzgvzCJyPFI5htuQzjW:-KzgvzCJyPFI5htuQzjX:b-16vsve) (no related file):*
I suggest we merge #3028 first, and then apply this change on top of that one.

@asashour can you regenerate the component files  for this change and push them together?

To recap, just run this command from the `flow-generated-components` module:

```
mvn clean install -Pgenerator
```

Thanks!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3018)*

**Comment by gilberto-torrezan** [comment]
:lgtm:

Most of the classes only had the `HasStyle` swapped with `ComponentSupplier`, but that's expected.

---

Reviewed 52 of 52 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/flow/3018#-:-KzhM2sjLOJj9ltmv5UX:bx4z5b6)*
